api.yaml

```diff
@@ -138,6 +140,46 @@ paths:
             schema:
               $ref: '#/components/schemas/link_accounts'
         required: true
+  /users/role:
+    post:
+      tags:
+        - admins
+        - noroles
+      summary: Modifies the role of the logged user.
+      operationId: setRole
+      description: |
+        Receives the new role to be added to the user
+      responses:
+        '200':
+          description: Role updated
+        '400':
+          description: 'invalid input, object invalid'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/set_role_body'
+        required: true
+  /users/events:
+    post:
+      summary: Receives the events of the SSO to sync the users
+      operationId: eventUser
+      description: |
+        Receives the SSO events and updates the users, both creating
+        the users and updating their roles and permissions 
+      responses:
+        '200':
+          description: Event processed
+        '400':
+          description: 'invalid input, object invalid'
+        '401':
+          description: 'One or both of the JWT are not valid'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
```

Comment:
A schema is required to document the precise definition of the body. This method looks like to me to a webhook. Maybe is wise, to open the scope of this endpoint to be a webhook instead of just to be used in a specific use case.

Reply:
updated using a template schema because /event is not used